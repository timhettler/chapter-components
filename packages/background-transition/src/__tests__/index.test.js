import React from 'react';
import Component from '../';
import renderer from 'react-test-renderer';

it(`${Component.name} renders correctly`, () => {
  const tree = renderer
    .create(
      <Component
        images={[
          'https://images.pexels.com/photos/1842623/pexels-photo-1842623.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500',
          'https://images.pexels.com/photos/773471/pexels-photo-773471.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500',
          'https://images.pexels.com/photos/1279330/pexels-photo-1279330.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500',
        ]}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it(`${Component.name} renders correctly with no images`, () => {
  const tree = renderer.create(<Component images={[]} />).toJSON();
  expect(tree).toMatchSnapshot();
});
