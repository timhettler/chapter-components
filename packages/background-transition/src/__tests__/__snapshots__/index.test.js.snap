// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundTransition renders correctly 1`] = `
<div
  className="bgTransition"
>
  <div
    className="bgTransition__img bgTransition__img--active"
    style={
      Object {
        "backgroundImage": "",
        "transitionDuration": "3000ms",
      }
    }
  />
  <div
    className="bgTransition__img"
    style={
      Object {
        "backgroundImage": "",
        "transitionDuration": "3000ms",
      }
    }
  />
  <div
    className="bgTransition__img"
    style={
      Object {
        "backgroundImage": "",
        "transitionDuration": "3000ms",
      }
    }
  />
</div>
`;
